src/main/java/ch/wisv/areafiftylan/products/controller/OrderRestController.java

```diff
@@ -149,11 +132,31 @@ public Order getOrderById(@PathVariable Long orderId) {
     @RequestMapping(value = "/orders/{orderId}", method = RequestMethod.DELETE)
     @JsonView(View.OrderOverview.class)
     public ResponseEntity<?> removeFromOrder(@PathVariable Long orderId, @RequestBody @Validated TicketDTO ticketDTO) {
-        Order modifiedOrder = orderService.removeTicketFromOrder(orderId, ticketDTO);
+        Order modifiedOrder = orderService
+                .removeTicketFromOrder(orderId, ticketDTO.getType(), ticketDTO.hasPickupService(),
+                        ticketDTO.isCHMember());
         return createResponseEntity(HttpStatus.OK, "Ticket successfully removed from Order", modifiedOrder);
     }
 
     /**
+     * Assign a User to a previously anonymous Order. The currently assigned user is assigned to the order specified in
+     * the path. This is required before an order can be paid.
+     *
+     * @param auth    The Spring Authentication object
+     * @param orderId Id of the Order to be assigned to the currently logged in user
+     *
+     * @return The assigned order
+     */
+    @PreAuthorize("isAuthenticated() and @currentUserServiceImpl.canAccessOrder(principal, #orderId)")
+    @PostMapping(value = "/orders/{orderId}/assign")
+    @JsonView(View.OrderOverview.class)
+    public ResponseEntity<?> assignOrderToUser(Authentication auth, @PathVariable Long orderId) {
+        Order assignedOrder = orderService.assignOrderToUser(orderId, auth.getName());
+        return createResponseEntity(HttpStatus.OK, "Order successfully attached to User", assignedOrder);
```

Comment:
Wouldn't it be better to return the updated order here? Changes are made to the order changing behavior with a POST request. It would seem logical to me that I get the updated order as a response.